NLP/top_comments.py

```diff
@@ -0,0 +1,95 @@
+import re
+import html
+import time
+import threading
+import streamlit as st
+from requests import get
+import pandas as pd
+
+BASE_URL = "https://hacker-news.firebaseio.com/v0/item/"
+CLEANR = re.compile('<.*?>')
+
+
+def get_comment_ids(story: int) -> list:
+    story_info = get(BASE_URL + f"{story}.json", timeout=30).json()
+    comment_ids = story_info.get("kids")
+    return comment_ids
+
+
+def format_html(text_string: str):
+    unescaped_text = html.unescape(text_string)
+    clean_text = re.sub(CLEANR, '', unescaped_text)
+    return clean_text
+
+
+def get_top_5_most_replied_parent_comments(story_id: int):
+    parent_comments = get_comment_ids(story_id)
+
+    parent_comments_list = []
+    for parent_comment_id in parent_comments:
+        comment_info = get(
+            BASE_URL + f"{parent_comment_id}.json", timeout=30).json()
+        comment_text = comment_info.get("text")
+        comment_title = format_html(
+            comment_text) if comment_text else "No title available"
+        number_of_children = 0
+        kids_info = comment_info.get("kids")
+        if kids_info is not None:
+            number_of_children = len(kids_info)
+
```

Comment:
Can you think of any ways to simplify and speed up this piece of code? The current runtime is quite long!